golang/pkg/builder/container/utils.go

```diff
@@ -138,15 +138,31 @@ func pullImage(ctx context.Context, logger io.StringWriter, fullyQualifiedImageN
 }
 
 func deleteContainer(ctx context.Context, containerName string) error {
-	if err := stopContainer(ctx, containerName); err != nil {
-		return err
+	deletableContainer, err := getContainer(ctx, containerName)
+	if err != nil {
+		return fmt.Errorf("builder could not get container (%s) to remove: %s", containerName, err.Error())
 	}
 
-	if err := removeContainer(ctx, containerName); err != nil {
-		return err
+	switch deletableContainer.State {
+	case "":
```

Comment:
Putting this before default would be better, IMHO.

Reply:
fixed